### ISSUE TITLE
[VFO] Scanning resets to where you started.

### ISSUE THREAD

**Issue by forkoz** [rogerclarkmelbourne/Radioddity_GD-77#31]
The most annoying thing about scanning on the GD-77 is that it resets to where you started as soon as the scan ends. If you catch a signal, you cannot stop to monitor the frequency but must remember and type it in. Seems like something that could be fixed with a NOP or two. Any idea which functions trigger when you enable scan mode?

**Comment by rogerclarkmelbourne** [comment]
OK.

That definitely sounds like one we can look at..

I don’t use scanning, so I will need to setup a scan list in the CPS and see what telltale triggers we can use to work out when it’s started and when it ends.

Actually, one thing we can do is look at what calls the function which changes the frequency.
It must be called, at least indirectly by the scanning code.

Also, we know which memory location holds the current TG, and that would be changed if the radio was in DMR mode

If we can find a hit list of functions, or ROM locations, I can set breakpoints in the code and see what gets triggered

**Comment by forkoz** [comment]
You can just scan on analog VFO, its less of a problem... but stilll a problem if you have a list. Hit list is the problem. Might be easier going from where the frequency is stored in ram.

**Comment by rogerclarkmelbourne** [comment]
OK. 

I setup a scan list on my GD-77, and I see what you mean.

As soon as the transmission stops, it seems to restart scanning from the first channel in the list.

Probably ideal functionality, would be to hold on the channel that has the signal, even after the transmission has ended.

This is probably achievable by removing something from the code.

Whats probably ideal, is that it holds on the channel for a given period of inactivity, and after that, it starts the scan again.

But I think thats a can of worms, because the timer needs to be cleared if you do something like manually change channel etc etc.


Strangely, if press the PTT, it doesnt seem to stop the scanning once you've released the PTT.

And...
If you use the CPS while its scanning, the scan function continues to run.

The CPS upload, should definitely stop the scan.

**Comment by forkoz** [comment]
Red menu key does stop the scan. Right arrow changes to VFO mode and also stops the scan in that mode. I scan from a list and scan in VFO for stuff unlisted. Would be cool to add color and slot searching in DMR VFO scan mode too, but that's getting a little ahead.

Right now:
Scan from list: red stops, arrow does nothing. resets to start on end
Analog VFO: Right arrow changes before/after dot, left arrow stop, red ends. resets to start
Digital VFO:  Right arrow changes before/after dot, left arrow stops, red ends, resets to start, kinda useless since everyone will use different DMR settings and I don't know if it stops on any signal or DMR only.

Where I am, analog & p.25 is used 10x more than DMR. Hang time is already settable in the CPS for all scan modes.

**Comment by rogerclarkmelbourne** [comment]
The scan is definitely not stopping for me when I press the red menu key or the orange button (which I have assigned to access the zones)

The only way to stop a scan is to press SK1 which i have assigned as the scan key

**Comment by forkoz** [comment]
No. Red doesn't stop lists either it looks like. Up or down does.
